crates/core/tedge_watchdog/src/systemd_watchdog.rs

```diff
@@ -149,23 +150,45 @@ async fn monitor_tedge_service(
     res_topic: Topic,
     interval: u64,
 ) -> Result<(), WatchdogError> {
-    let client_id: &str = &format!("{}_{}", name, nanoid!());
     let config_repository = tedge_config::TEdgeConfigRepository::new(tedge_config_location);
     let tedge_config = config_repository.load()?;
+
+    let mqtt_device_topic_id: EntityTopicId = tedge_config
+        .mqtt
+        .device_topic_id
+        .parse()
+        .context("Can't parse as device topic id")?;
+
+    let mqtt_topic_root = &tedge_config.mqtt.topic_root;
+
+    let mqtt_session_name = format!("{SERVICE_NAME}#{mqtt_topic_root}/{mqtt_device_topic_id}");
+
+    let mqtt_schema = MqttSchema::with_root(mqtt_topic_root.clone());
+
+    let service_health_topic =
+        ServiceHealthTopic::from_new_topic(&mqtt_device_topic_id.into(), &mqtt_schema);
```

Comment:
```suggestion
    let service_topic_id = mqtt_device_topic_id
        .default_service_for_device(SERVICE_NAME)
        .unwrap();
    let service_health_topic =
        ServiceHealthTopic::from_new_topic(&service_topic_id.into(), &mqtt_schema);
```
That doesn't look right. You're creating a service health topic from the device's `device_topic_id` directly. Don't you have to add the service name: `tedge-watchdog` here?

Reply:
Fixed bcd2633e83